crates/matrix-sdk/src/room/timeline/event_item.rs

```diff
@@ -258,6 +258,9 @@ pub enum TimelineItemContent {
 
     /// A redacted message.
     RedactedMessage,
+
+    /// An `m.room.encrypted` event that could not be decrypted.
+    UnableToDecrypt,
```

Comment:
We know nothing about the content that we would like to "present" to SDK users here, right?

Reply:
don't we have like the sender at least? And some other metadata like the time it was sent? That's at least what element-web shows...

Reply:
You don't know anything about the thing that's encrypted, if you mean the content type or some other metadata.

Though you will want to show quite a bit, you'll likely want to have some "request room key functionality", what springs to mind, besides the sender and timestamp, is:

* Algorithm that the event claims to use
* Session id
* `sender_key` and Device id if there is one

Hmm, that's almost the whole content of the event, perhaps it makes sense to leave the whole content in.

Reply:
@gnunicorn Sender and time are not part of the content, they are fields of the timeline item (one level up basically).

Reply:
Since we store the raw event separately as well and probably need to deserialize to `matrix-sdk-crypto`'s `RoomEncryptedEventContent`, I'd rather introduce a payload-free version of Ruma's `RoomEncryptedEventContent` to use here. The encrypted payload isn't going to be very useful outside of show-event-source, if at all.